### PR TITLE
gh-107901: duplicate blocks with no lineno that have an eval break and multiple predecessors

### PR DESCRIPTION
**Pull request by iritkatriel**
This ensures that these blocks get a line number propagated to them.

* Issue: gh-107901

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2024-01-11-16-54-55.gh-issue-107901.Td3JPI.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Compiler duplicates basic blocks that have an eval breaker check, no line number, and multiple predecessors on different lines.
```
Comment (iritkatriel) [inline Misc/NEWS.d/next/Core and Builtins/2024-01-11-16-54-55.gh-issue-107901.Td3JPI.rst]:
```suggestion
Compiler duplicates basic blocks that have an eval breaker check, no line number, and multiple predecessors.
```

File: Python/flowgraph.c [hunk 2]
```diff
@@ -2246,8 +2256,8 @@ convert_pseudo_ops(basicblock *entryblock)
 }
 
 static inline bool
-is_exit_without_lineno(basicblock *b) {
-    if (!basicblock_exits_scope(b)) {
+is_exit_or_eval_check_without_lineno(basicblock *b) {
+    if (! (basicblock_exits_scope(b) || basicblock_has_eval_break(b))) {
```
Comment (markshannon) [inline Python/flowgraph.c]: Negations around ands/ors are a bit hard to follow. Maybe add another if block for `!basicblock_has_eval_break(b)` after this one?

**Review by markshannon — COMMENTED** [review]
One small suggestion, otherwise LGTM.
